static/js/components/widgets/SelectField.tsx

```diff
@@ -1,36 +1,57 @@
-import React, { useCallback, ChangeEvent } from "react"
+import React, { useCallback, ChangeEvent, useState } from "react"
 import Select from "react-select"
-import AsyncSelect from "react-select/async"
 import { isNil } from "ramda"
+import { AsyncPaginate, LoadOptions } from "react-select-async-paginate"
+import AsyncSelect from "react-select/async"
+import { isFeatureEnabled } from "../../util/features"
 
 export interface Option {
   label: string
   value: string
 }
 
+export interface Additional {
+  callback?: (options: Option[]) => void
+}
+
 interface Props {
   name: string
-  value: null | undefined | string | string[]
+  classNamePrefix?: string
+  value?: null | undefined | string | string[]
   onChange: (event: ChangeEvent<HTMLSelectElement>) => void
   multiple?: boolean
   options: Array<string | Option>
-  loadOptions?: (s: string, cb: (options: Option[]) => void) => void
+  loadOptions?: LoadOptions<Option, Option[], Additional | undefined>
   placeholder?: string
   defaultOptions?: Option[]
+  cacheUniques?: ReadonlyArray<any>
+  hideSelectedOptions?: boolean
+  preserveSearchText?: boolean
   isOptionDisabled?: (option: Option) => boolean
+  isOptionSelected?: (option: Option) => boolean
 }
 
 export default function SelectField(props: Props): JSX.Element {
   const {
     value,
     onChange,
     name,
+    classNamePrefix,
     options,
     loadOptions,
     defaultOptions,
-    placeholder,
-    isOptionDisabled
+    placeholder: initialPlaceholder,
+    isOptionDisabled,
+    isOptionSelected,
+    hideSelectedOptions,
+    preserveSearchText = false,
+    cacheUniques
   } = props
+  const [searchText, setSearchText] = useState("")
+  const [placeholder, setPlaceholder] = useState("")
```

Comment:
I noticed a small UI bug with `SortableSelect`. When `FEATURE_SORTABLE_SELECT_QUICK_ADD` is in use, 
1. Search for "3play" (or anything)
2. Close the menu, so `placeholder = "3play"`
3. Open the search menu, and clear it.

The `"3play"` placeholder is still shown:

<img width="200" alt="Screenshot 2023-08-14 at 11 51 13 AM" src="https://github.com/mitodl/ocw-studio/assets/9010790/626256eb-fe6a-42fc-9c77-d4f22f813006">

**Comment / Question:**  Is the placeholder state value necessary? Is there ever a time placeholder and searchtext are intentionally different?

Reply:
I think I understand what's happening here. I'll fix that.

> Is the placeholder state value necessary?

Yes. When the field is unfocused, `Select` hides the input element. Which hides the search text. I tried handling it through styling but that didn't work. Handling it through the styles also causes a problem with showing selected values.

Basically, this is what happens:

https://github.com/mitodl/ocw-studio/assets/71316217/1dbab922-20ac-488b-a639-e4ad037488a1

>  Is there ever a time placeholder and searchtext are intentionally different?

Yes, the placeholder and the search text are different when the field is unfocused. More specifically the `searchText` is `""` when the field is unfocused.

The `Select` field has only one area to show the following three types of values (in order of presendence)

1. Search text (`inputValue` prop)
2. Field value
3. Placeholder

If we don't set `searchText=""`, the field won't show the field value. So we use `placeholder` as a backup **and** as a way to show the search text when nothing is selected.

Reply:
> The Select field has only one area to show the following three types of values (in order of presendence)
> 
> 1. Search text (inputValue prop)
> 2. Field value
> 3. Placeholder
> If we don't set searchText="", the field won't show the field value. So we use placeholder as a backup and as a way to show the search text when nothing is selected.

Thanks, great explanation. I think I was confused because SortableSelects never show the field value in the select field (right?); other selects, do, though. 

Anyway, that makes sense 👍